Sources/UI/PaymentMethods/Views/InvoiceLoginView.swift

```diff
@@ -0,0 +1,178 @@
+                },
+                header: {
+                    Text(LoginStrings.info.localizedString(domain))
+                }
+            )
+            .textCase(nil)
+            Section(
+                content: {
+                    button
+                },
+                footer: {
+                    if !loginModel.errorMessage.isEmpty {
+                        Text(keyed: loginModel.errorMessage)
+                            .font(.footnote)
+                            .foregroundColor(.red)
+                    }
+                })
+        }
+        .onChange(of: loginModel.isValid) { isValid in
+            if model.isWaiting {
+                canLogin = false
+            } else {
+                canLogin = isValid
+            }
+        }
+    }
+}
+
+public struct InvoiceDetailView: View {
+    @ObservedObject private var loginModel: InvoiceLoginModel
+    @Environment(\.presentationMode) var presentationMode
+    let domain = "Snabble.Payment.ExternalBilling"
+
+    public init(model: InvoiceLoginModel) {
+        self.loginModel = model
+    }
+        
+    public var body: some View {
+        Form {
+            Section(
+                content: {
+                    HStack {
+                        if let image = loginModel.image {
+                            image
+                        }
+                        Text(loginModel.username)
+                    }
+                },
+                header: {
+                    Text(LoginStrings.username.localizedString(domain))
+                },
+                footer: {
+                    Text(keyed: "Snabble.Payment.ExternalBilling.hint")
+                })
+            .textCase(nil)
+        }
+        .toolbar {
+            ToolbarItem(placement: .destructiveAction) {
+                Button(action: {
+                    loginModel.actionPublisher.send(["action": LoginViewModel.Action.remove.rawValue])
+                }) {
+                    Image(systemName: "trash")
+                }
+                .foregroundColor(Color.accent())
+            }
+        }
+    }
+}
+
+public struct InvoiceView: View {
+    @ObservedObject var loginProcessor: InvoiceLoginProcessor
+    @StateObject private var loginModel: InvoiceLoginModel
+    
+    init(model: InvoiceLoginProcessor) {
+        self.loginProcessor = model
+        self._loginModel = StateObject(wrappedValue: model.invoiceLoginModel)
+
+    }
+    @ViewBuilder
+    var content: some View {
+        if loginModel.paymentUsername != nil || loginModel.isLoggedIn {
+            InvoiceDetailView(model: loginModel)
+        } else {
+            InvoiceLoginView(model: self.loginProcessor)
+        }
+    }
+
+    public var body: some View {
+        content
+            .onChange(of: loginModel.isLoggedIn) { loggedIn in
+                if loggedIn, let info = loginModel.loginInfo {
+                    print("user is logged in \(info))")
```

Comment:
It should be removed

Reply:
cadabf76314400957aa01801d969f59a559ee8cd